### PR TITLE
Set correct proxy mode for API request

### PR DESCRIPTION
**Pull request by peceka**
Adding forgotten variable proxy_mode.

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @peceka, thanks for the PR. Can you take a look at the rubocop issue? There is one wrong whitespace. Are you able to add tests for the change?

**Comment by peceka** [comment]
@bastelfreak I will correct rubocop issue. With tests I will read about them and try to do this as well.

**Comment by bastelfreak** [comment]
@peceka let us know if you need any help for tests. Feel free to join our IRC channel #voxpupuli on freenode if you have any questions.

File: lib/puppet/provider/zabbix_proxy/ruby.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@ def create
 
     # Set some vars
     host = @resource[:hostname]
+    mode = @resource[:mode]
```
Comment (flaupretre) [inline lib/puppet/provider/zabbix_proxy/ruby.rb]: I had the same problem when installing a zabbix server. This patch solved it but I had to change line 12 to 'mode = @resource[:mode].to_i'.

**Comment by bastelfreak** [comment]
@peceka can you take a look at the failing spec tests?

**Comment by Fabian1976** [comment]
I also ran into this error (no implicit conversion of Fixnum into String), but changing this line:
```proxy_mode = @resource[:mode] + 5```
to this:
```proxy_mode = @resource[:mode].to_i + 5```

fixed it.

**Comment by bastelfreak** [comment]
Hi people. I'm going to close this PR due to inactivity. Please reopen it or resubmit it if one of you wants to continue working on it.

**Comment by Fabian1976** [comment]
To bad it is closed without fixing the issue!

**Pull request closed without merging**